protos/flyteidl/event/event.proto

```diff
@@ -62,8 +62,20 @@ message NodeExecutionEvent {
         WorkflowNodeMetadata workflow_node_metadata = 8;
     }
 
-    // Specifies which task (if any) launched this node.
+    // [To be deprecated] Specifies which task (if any) launched this node.
     ParentTaskExecutionMetadata parent_task_metadata         = 9;
+
+    // Specifies the parent node of the current node execution. Node executions at level zero will not have a parent node.
+    ParentNodeExecutionMetadata parent_node_metadata	= 10;
+
+    // Group identifier to indicate arbitrary grouping like retries, sub workflow
+    string group_id = 11;
```

Comment:
question: should this be part of the node execution identifier?

Reply:
Unique identifier of the node is the NodeExecutionId - this is just a label/id to group node executions.

Reply:
is this label unique within the execution?

Reply:
@katrogan It's not - as in need not be. It only need to be unique within a parent node. But that's up to implementation, and future use cases.

Reply:
so effectively, there would be no enforcement or semantic checking on this field?

Reply:
but If I want to query all node-executions for a parent node, then I could get the parent node and find the group id?

Reply:
@kumare3 What do you mean semantic checking? 

If you want all executions for a parent node, you will simply pass the filter with the value of unique_node_id in it. Group id is for retries inside the parent node.